client/client.go

```diff
@@ -226,7 +265,19 @@ func NewClientTLSConfig(creds *identity.Credentials, authMethod identity.AuthMod
 		option(opts)
 	}
 
-	return newClient(creds, authMethod, goosehttp.NewWithTLSConfig(config, goosehttp.WithHeadersFunc(opts.httpHeadersFunc)), logger)
+	client := *opts.httpClient
+	if client.Transport == nil {
+		client.Transport = &http.Transport{}
+	}
+	if t, ok := client.Transport.(*http.Transport); ok {
+		t.TLSClientConfig = config
```

Comment:
We should return an error if ok false. Otherwise you'd get a client without a TLS config and not know it.